internal/pkg/config/output_test.go

```diff
@@ -382,3 +382,170 @@ func setTestEnv(t *testing.T, env map[string]string) {
 		t.Setenv(k, v)
 	}
 }
+
+func TestMergeElasticsearchFromPolicy(t *testing.T) {
+	cfg := Elasticsearch{
+		Protocol:         "http",
+		Hosts:            []string{"elasticsearch:9200"},
+		ServiceToken:     "token",
+		Timeout:          time.Second,
+		MaxRetries:       1,
+		MaxConnPerHost:   1,
+		MaxContentLength: 1,
+	}
+	tests := []struct {
+		name string
+		pol  Elasticsearch
+		res  Elasticsearch
+	}{{
+		name: "default policy",
+		pol: Elasticsearch{
+			Hosts:            []string{"localhost:9200"},
+			Timeout:          DefaultElasticsearchTimeout,
+			MaxRetries:       DefaultElasticsearchMaxRetries,
+			MaxConnPerHost:   DefaultElasticsearchMaxConnPerHost,
+			MaxContentLength: DefaultElasticsearchMaxContentLength,
+		},
+		res: Elasticsearch{
+			Protocol:         "http",
+			Hosts:            []string{"elasticsearch:9200"},
+			ServiceToken:     "token",
+			Timeout:          time.Second,
+			MaxRetries:       1,
+			MaxConnPerHost:   1,
+			MaxContentLength: 1,
+		},
+	}, {
+		name: "hosts differ",
+		pol: Elasticsearch{
+			Protocol:         "https",
+			Hosts:            []string{"elasticsearch:9200", "other:9200"},
+			Timeout:          DefaultElasticsearchTimeout,
+			MaxRetries:       DefaultElasticsearchMaxRetries,
+			MaxConnPerHost:   DefaultElasticsearchMaxConnPerHost,
+			MaxContentLength: DefaultElasticsearchMaxContentLength,
+		},
+		res: Elasticsearch{
+			Protocol:         "https",
+			Hosts:            []string{"elasticsearch:9200", "other:9200"},
+			ServiceToken:     "token",
+			Timeout:          time.Second,
+			MaxRetries:       1,
+			MaxConnPerHost:   1,
+			MaxContentLength: 1,
+		},
+	}, {
+		name: "all non tls attributes differ",
+		pol: Elasticsearch{
+			Protocol:         "https",
+			Hosts:            []string{"elasticsearch:9200", "other:9200"},
+			Headers:          map[string]string{"custom": "value"},
+			ProxyURL:         "http://proxy:8080",
+			ProxyDisable:     false,
+			ProxyHeaders:     map[string]string{"proxyhead": "proxyval"},
+			Timeout:          time.Second * 2,
+			MaxRetries:       2,
+			MaxConnPerHost:   3,
+			MaxContentLength: 4,
+		},
+		res: Elasticsearch{
+			Protocol:         "https",
+			Hosts:            []string{"elasticsearch:9200", "other:9200"},
+			Headers:          map[string]string{"custom": "value"},
+			ProxyURL:         "http://proxy:8080",
+			ProxyDisable:     false,
+			ProxyHeaders:     map[string]string{"proxyhead": "proxyval"},
+			ServiceToken:     "token",
+			Timeout:          2 * time.Second,
+			MaxRetries:       2,
+			MaxConnPerHost:   3,
+			MaxContentLength: 4,
+		},
+	}}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			res := MergeElasticsearchFromPolicy(cfg, tc.pol)
```

Comment:
@michel-laterman why do we need to merge the policy here? cannot use what come from the config? the proxy and tls settings should be configured there too no? otherwise there is no way for user to change `proxy` and `tls` from Fleet right?

Reply:
Currently the elastic-agent injection will not replace keys that are already present in local config:

https://github.com/elastic/elastic-agent/blob/2873f66e40fbbb490dc1367cb4cf2c9ccc937ae3/internal/pkg/agent/application/fleet_server_bootstrap.go#L67-L77

So the agent can set `tls.CAs` from policy if it was not an enrollment arg (i'm not sure if it serializes locally)
but the bigger issue is that the agent only ever sends a single host in the output block due to this behaviour